Unable to import azurerm_kubernetes_cluster

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.7

### AzureRM Provider Version

3.39.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "3.39.0"
    }
  }
}
provider "azurerm" {
  features {}
}
```


### Debug Output/Panic Output

```shell
│ Error: Plugin did not respond
│ 
│   with azurerm_kubernetes_cluster.aks,
│   on main.tf line 59, in resource "azurerm_kubernetes_cluster" "aks":
│   59: resource "azurerm_kubernetes_cluster" "aks" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

**TRACE log**

AzureRM Request:
GET /subscriptions/0000-000000-0000-000000000000/resourceGroups/rg000/providers/Microsoft.ContainerService/managedClusters/cluster000?api-version=2022-09-
02-preview HTTP/1.1
Host: management.azure.com
{
  "id": "/subscriptions/0000-000000-0000-000000000000/resourceGroups/rg000/providers/Microsoft.ContainerService/managedClusters/cluster000",    
  "type": "Microsoft.ContainerService/ManagedClusters",
  "properties": {     
          **"autoUpgradeProfile": {},**     
          "workloadAutoScalerProfile": {}
}

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4c12582]

goroutine 216 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterRead( )
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:2060 +0x9c2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read( )
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:712 +0x178
```


### Expected Behaviour

It should be possible to import AKS cluster with `"autoUpgradeProfile": {}`.

### Actual Behaviour

Unable to import AKS cluster.
This looks like an issue during API response processing.
I don't know go and am not able to create PR with fix for that case. 
Please advice.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -2096,10 +2096,8 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{}) 
 		d.Set("public_network_access_enabled", publicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
 
 		upgradeChannel := ""
-		if profile := props.AutoUpgradeProfile; profile != nil {
-			if v := profile.UpgradeChannel; v != nil {
-				upgradeChannel = string(*profile.UpgradeChannel)
-			}
+		if profile := props.AutoUpgradeProfile; profile != nil && profile.UpgradeChannel != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {
+			upgradeChannel = string(*profile.UpgradeChannel)
 		}
 		d.Set("automatic_channel_upgrade", upgradeChannel)
 

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -2096,8 +2096,10 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{}) 
 		d.Set("public_network_access_enabled", publicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
 
 		upgradeChannel := ""
-		if profile := props.AutoUpgradeProfile; profile != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {
-			upgradeChannel = string(*profile.UpgradeChannel)
+		if profile := props.AutoUpgradeProfile; profile != nil {
+			if v := profile.UpgradeChannel; v != nil {
+				upgradeChannel = string(*profile.UpgradeChannel)
+			}
 		}
 		d.Set("automatic_channel_upgrade", upgradeChannel)
 

